WordPress/gutenberg
# [Mobile] Moves block settings and more button to the toolbar

**Related PRs:**
- https://github.com/wordpress-mobile/gutenberg-mobile/pull/5933

## What?
This PR moves the block mobile toolbar and moves the settings and more buttons to the header toolbar.

## Why?
Part of the design changes project to make the mobile editor more consistent with the Web editor.

## How?

- It removes the BlockMobileToolbar and moves some of its content to the BlockToolbar
- Updates the BlockDraggable tests since the BlockMobileToolbar was removed
- Updates the BlockMover component to add a ToolbarGroup wrapper
- Updates the BlockSettingsButton component and simplifies it since it is now within the block toolbar, no workarounds are needed.
- Removed logic to wrap icons, this was done before when the BlockMobileToolbar existed within the block's content, for cases where the block wrapper would be small, e.g. Buttons, Social links blocks.
- BlockToolbar is updated to handle delete block logic and the `isStackedHorizontally` prop. It also now renders the BlockSettingsButton, BlockMover, and BlockActionsMenu.
- EditorPage: it adds a new helper `swipeToolbarToElement` to be able to scroll within the toolbar for cases of removing a block, adding a block, and toggling formatting buttons.
- Updates tests.

## Testing Instructions
- Open the app
- Add a block
- **Expect** to see the SettingsButton, and the More button within the header toolbar
- Add a Gallery block and some images
- **Expect** to see the left/right arrow buttons to move the blocks within the toolbar
- Add several blocks
- Expect to see the up/down buttons within the toolbar

### Testing Instructions for Keyboard
N/A

## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/4885740/a74257cd-0b15-43df-9fc5-0212b314bfb2
